src/openrct2/rct12/ScenarioPatcher.cpp

```diff
@@ -93,7 +104,33 @@ static u8string ToOwnershipJsonKey(int ownershipType)
     return {};
 }
 
-static std::vector<TileCoordsXY> getCoordinates(const json_t& parameters)
+static void readCoordinate(std::vector<TileCoordsXY>& out, const json_t& coordinatesArray)
+{
+    if (coordinatesArray.size() != 2)
+    {
+        OpenRCT2::Guard::Assert(0, "Fix coordinates sub array should have 2 elements");
+        return;
+    }
+
+    out.emplace_back(
+        OpenRCT2::Json::GetNumber<int32_t>(coordinatesArray[0]), OpenRCT2::Json::GetNumber<int32_t>(coordinatesArray[1]));
+}
+
+static void readCoordinate(std::vector<TileCoordsXYZ>& out, const json_t& coordinatesArray)
+{
+    if (coordinatesArray.size() != 3)
+    {
+        OpenRCT2::Guard::Assert(0, "Fix coordinates sub array should have 3 elements");
```

Comment:
It's a shame we use asserts for such conditions. There should be an error displayed with link to relevant source file and line - not part of this PR.